pandas/tools/pivot.py

```diff
@@ -436,18 +443,56 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    if values is None and aggfunc is not None:
+        raise ValueError("`aggfunc` cannot be used with `values`.")
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+        table = table.fillna(0).astype(np.int64)
+
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=aggfunc, margins=margins, dropna=dropna)
-        return table
+
+    # Post-process
+    table = _normalize(table, normalize=normalize, margins=margins)
+
+    return table
+
+
+def _normalize(table, normalize, margins):
+
+    # Actual Normalizations
+    normalizers = {
+        False: lambda x: x,
+        'all': lambda x: x / x.sum(axis=1).sum(axis=0),
+        'columns': lambda x: x / x.sum(),
+        'index': lambda x: x.div(x.sum(axis=1), axis=0)
+
+    }
+
+    normalizers[True] = normalizers['all']
+
+    try:
+        f = normalizers[normalize]
+    except KeyError:
+        raise ValueError("Not a valid `normalize` argument")
+
+    table = f(table)
+
+    if margins is True and normalize is not False:
+        table = table * 2
+        if normalize == "index":
+            table = table.drop('All', axis=1)
+        if normalize == "columns":
+            table = table.drop('All', axis=0)
```

Comment:
multiple ifs should either be if/else, or function tables. Function tables are more general and that's how we do it in the code-base.